### DISCUSSION TITLE
How to disable swagger and enable redoc for one endpoint ?

### DISCUSSION THREAD

**Discussion by Courouge** [fastapi/fastapi#2655] [Questions] [answered]
Hey all,

I am wondering if it is possible to disable swagger url and enable redoc url for one endpoint?

Globaly it is possible with:
app = FastAPI(docs_url=None) # redoc_url in default /redoc

The only thing I found is to disable both swagger and redoc :
@app.get("/test", include_in_schema=False) # well I need /test in redoc url

And it is not possible in router...
router(docs_url=None)

How can I handle this case ?

**Comment by SnkSynthesis** [comment, marked as answer]
@Courouge,

You can look into the [Extending OpenAPI](https://fastapi.tiangolo.com/advanced/extending-openapi/) section of the docs and you could have the default OpenAPI schema URL for ReDoc and another separate customized OpenAPI schema URL for Swagger.

Or you can use [Sub Applications](https://fastapi.tiangolo.com/advanced/sub-applications/) and disable `/docs` for the sub app and instead only enable `/redoc`.
